### PR TITLE
ntheory: divisor_sigma symbolic case recognition

### LINKED ISSUE #16123: symbolic divisor_sigma
`divisor_sigma(a**k*b**j, 1)` where `a` and `b` are prime and `k` and `j` are positive integers is `(a**(k+1)-1)/(a-1)*(b**(j + 1) -1)/(b - 1)`. Right now, the only case that is recognized symbolically is `divisor_sigma(a) -> a + 1`.
```python
def eval(cls, n):
    ...
    if n.is_integer:
        if n.is_Mul or n.is_Pow:  # prime is already handled
            p = []
            for b, e in (_.as_base_exp() for _ in Mul.make_args(n)):
                if b.is_prime and e.is_positive:
                    p.append((b**(e + 1) - 1)/(b - 1))
                else:
                    return
            return Mul(*p)

>>> divisor_sigma(2**3*3**5)
5460
>>> a,b = var('a b',prime=True)
>>> j,k=var('j k', integer=True, positive=True)
>>> eq = a**j*b**k
>>> eq.subs({a:2,j:3,b:3,k:5})
5460
```

### PR DESCRIPTION
**Pull request by sachin-4099**
Fixes: #16123

#### Brief description of what is fixed or changed
**Previously**, the only case which was recognized symbolically by divisor_sigma() function was: `divisor_sigma(a) -> a + 1.`

**Now**, it also recognizes:
`divisor_sigma(a**j*b**k)` where` a and b are prime` and` j and k are positive integers`
`divisor_sigma(a**j*b**k)-> (a**(j + 1) - 1)*(b**(k + 1) - 1)/((a - 1)*(b - 1))`

#### Other Comments
Regression Tests have been added

#### Release Notes

* ntheory
  * Made divisor_sigma() symbolically recognize the case: `divisor_sigma(a**j*b**k)`

### CODE REVIEW

**Comment by smichr** [comment]
I think you are only considering the case for `k==1`:
```python
>>> divisor_sigma(n*m,2).subs(((n,3),(m,5))),divisor_sigma(15,2)
(24, 260)
>>> divisor_sigma(n*m,1).subs(((n,3),(m,5))),divisor_sigma(15,1)
(24, 24)
>>> divisor_sigma(n*m,0).subs(((n,3),(m,5))),divisor_sigma(15,0)
(24, 4)
```

**Comment by sachin-4099** [comment]
@smichr please review.

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -2039,8 +2039,20 @@ class divisor_sigma(Function):
...
         n = sympify(n)
         k = sympify(k)
+
         if n.is_prime:
             return 1 + n**k
+
+        if n.is_integer and not n.is_Integer:
+            p = []
+            for b, e in (_.as_base_exp() for _ in Mul.make_args(n)):
+                if b.is_prime and e.is_positive:
```
Comment (czgdp1807) [inline sympy/ntheory/factor_.py]: If this fails then `None` will be returned? Is this expected? Any examples which can cause this to fail?

Comment (sachin-4099) [reply sympy/ntheory/factor_.py]: No that same expression will be returned and this case is possible:

```
In[1]: k, m, n = symbols('k m n', integer=True)
In[2]: divisor_sigma(k*m)
Out[2]: divisor_sigma(k*m)


```

File: sympy/ntheory/tests/test_factor_.py [hunk 2]
```diff
@@ -402,6 +402,12 @@ def test_divisor_sigma():
     assert divisor_sigma(23450, 2) == 730747500
     assert divisor_sigma(23450, 3) == 14666785333344
 
+    a = Symbol("a", prime=True)
+    b = Symbol("b", prime=True)
+    j = Symbol("j", integer=True, positive=True)
+    k = Symbol("k", integer=True, positive=True)
+    assert divisor_sigma(a**j*b**k) == (a**(j + 1) - 1)*(b**(k + 1) - 1)/((a - 1)*(b - 1))
```
Comment (smichr) [inline sympy/ntheory/tests/test_factor_.py]: You should add a test for `k == 0` and 2, also.

Comment (sachin-4099) [reply sympy/ntheory/tests/test_factor_.py]: No problem will add asap.

**Comment by smichr** [comment]
Looks good. Ready to commit once tests pass.

**Comment by sachin-4099** [comment]
@smichr ready to merge

**Pull request merged**